src/IssueViz.UnitTests/IssueVisualizationControl/IssueVisualizationViewModelTests.cs

```diff
@@ -395,30 +483,61 @@ public void SelectionService_OnIssueChanged_PreviousIssueIsNotNull_NewIssueIsNot
+
+            var newLocation = CreateLocation();
+            var newFlow = CreateFlow(newLocation);
+            var newIssue = CreateIssue(flows: new[] {newFlow});
+
+            Mock.Get(newIssue).SetupAdd(x => x.PropertyChanged += null);
+
+            RaiseSelectionChangedEvent(SelectionChangeLevel.Issue, newIssue, newFlow, newLocation);
+
+            Mock.Get(previousIssue).VerifyRemove(x => x.PropertyChanged -= It.IsAny<PropertyChangedEventHandler>(), Times.Once);
+            Mock.Get(newIssue).VerifyAdd(x => x.PropertyChanged += It.IsAny<PropertyChangedEventHandler>(), Times.Once);
+        }
+
+        [TestMethod]
+        public void SelectionService_OnIssueChanged_UnregisterFromPreviousLocationsAndRegisterToNewLocations()
         {
-            var previousIssue = new Mock<IAnalysisIssueVisualization>();
-            previousIssue.SetupAdd(x => x.PropertyChanged += null);
+            var previousLocation = CreateLocation();
+            var previousFlow = CreateFlow(previousLocation);
+            var previousIssue = CreateIssue(flows: new[] {previousFlow});
 
-            var previousFlow = CreateFlow(out var previousLocation);
-            RaiseSelectionChangedEvent(SelectionChangeLevel.Issue, previousIssue.Object, previousFlow, previousLocation);
+            Mock.Get(previousLocation).SetupAdd(x => x.PropertyChanged += null);
 
-            previousIssue.VerifyAdd(x=> x.PropertyChanged += It.IsAny<PropertyChangedEventHandler>(), Times.Once);
+            RaiseSelectionChangedEvent(SelectionChangeLevel.Issue, previousIssue, previousFlow, previousLocation);
+
+            Mock.Get(previousLocation).VerifyAdd(x => x.PropertyChanged += It.IsAny<PropertyChangedEventHandler>(), Times.Once);
 
-            var newIssue = new Mock<IAnalysisIssueVisualization>();
-            newIssue.SetupAdd(x => x.PropertyChanged += null);
+            var newLocation = CreateLocation();
+            var newFlow = CreateFlow(newLocation);
+            var newIssue = CreateIssue(flows: new[] {newFlow});
 
-            var newFlow = CreateFlow(out var newLocation);
-            RaiseSelectionChangedEvent(SelectionChangeLevel.Issue, newIssue.Object, newFlow, newLocation);
+            Mock.Get(newLocation).SetupAdd(x => x.PropertyChanged += null);
+
+            RaiseSelectionChangedEvent(SelectionChangeLevel.Issue, newIssue, newFlow, newLocation);
 
-            previousIssue.VerifyRemove(x => x.PropertyChanged -= It.IsAny<PropertyChangedEventHandler>(), Times.Once);
-            newIssue.VerifyAdd(x => x.PropertyChanged += It.IsAny<PropertyChangedEventHandler>(), Times.Once);
+            Mock.Get(previousLocation).VerifyRemove(x => x.PropertyChanged -= It.IsAny<PropertyChangedEventHandler>(), Times.Once);
+            Mock.Get(newLocation).VerifyAdd(x => x.PropertyChanged += It.IsAny<PropertyChangedEventHandler>(), Times.Once);
```

Comment:
General point about the tests working with collection of locations: in general there's only one item in the collection, so strictly speaking the test doesn't check whether the collection is being handled correctly; it could just be the first item.
It's worth thinking about whether it's worth testing with more than one item or not in each case since it's clearly more effort. In most cases it's fairly low risk to only test one item, but it is a gap in the functional coverage.